### PR TITLE
fix: [ANDROAPP-3147] View complete enrollment dashboard

### PR DESCRIPTION
**Pull request by Balcan**
## Description
[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3147)

## Solution description
Added new parameter enrollmentUid needed to open the dashboard.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [x] Smartphone Emulator
- [x] Tablet Emulator
- [x] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [x] 8.X
- [x] 9.X - 10.X
- [ ] Other
## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureFragment/EventCaptureFormFragment.java [hunk 1]
```diff
@@ -205,9 +205,9 @@ public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
     private void checkLastItem() {
         GridLayoutManager layoutManager = (GridLayoutManager) binding.formRecycler.getLayoutManager();
         int lastVisiblePosition = layoutManager.findLastVisibleItemPosition();
-        boolean shouldShowFab =
+        boolean shouldShowFab = lastVisiblePosition != -1 && (
                 lastVisiblePosition == dataEntryAdapter.getItemCount() - 1 ||
-                        dataEntryAdapter.getItemViewType(lastVisiblePosition) == 17;
+                        dataEntryAdapter.getItemViewType(lastVisiblePosition) == 17);
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureFragment/EventCaptureFormFragment.java:210]: Why 17? What does 17 means?

File: app/src/main/java/org/dhis2/usescases/teiDashboard/DashboardRepositoryImpl.java [hunk 2]
```diff
@@ -106,38 +104,33 @@ public Event updateState(Event eventModel, EventStatus newStatus) {
...
+                            .equals(d2.programModule().programs().uid(programUid).blockingGet().ignoreOverdueEvents()))
+                        if (event.status() == EventStatus.SCHEDULE
+                                && event.dueDate().before(DateUtils.getInstance().getToday()))
+                            event = updateState(event, EventStatus.OVERDUE);
+
+                    return event;
+                }).toSortedList((event1, event2) ->
+                        EventExtensionsKt.primaryDate(event2).compareTo(
+                                EventExtensionsKt.primaryDate(event1)
+                        ))
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/usescases/teiDashboard/DashboardRepositoryImpl.java:126]: Can we use the new `.orderByTimelime()`'s sdk method in here??

File: app/src/main/java/org/dhis2/usescases/teiDashboard/StatusChangeResultCode.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.dhis2.usescases.teiDashboard
+
+enum class StatusChangeResultCode {
+    CHANGED,
+    FAILED,
+    ACTIVE_EXIST,
+    WRITE_PERMISSION_FAIL
+}
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/usescases/teiDashboard/StatusChangeResultCode.kt:8]: 👏

File: app/src/main/java/org/dhis2/usescases/teiDashboard/TeiDashboardMobileActivity.java [hunk 4]
```diff
@@ -481,7 +481,8 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             }
 
             if (data.hasExtra("CHANGE_PROGRAM")) {
-                startActivity(intent(this, teiUid, data.getStringExtra("CHANGE_PROGRAM"), null));
+                startActivity(intent(this, teiUid, data.getStringExtra("CHANGE_PROGRAM"),
+                        data.getStringExtra("CHANGE_PROGRAM_ENROLLMENT")));
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/usescases/teiDashboard/TeiDashboardMobileActivity.java:485]: We could have Constant string values in here.

File: app/src/main/java/org/dhis2/usescases/teiDashboard/teiProgramList/TeiProgramListActivity.java [hunk 5]
```diff
@@ -97,11 +97,12 @@ public void goToEnrollmentScreen(String enrollmentUid, String programUid) {
     }
 
     @Override
-    public void changeCurrentProgram(String program) {
+    public void changeCurrentProgram(String program, String enrollmentUid) {
         if (program != null)
             SetProgramTheme(presenter.getProgramColor(program));
         Intent data = new Intent();
         data.putExtra("CHANGE_PROGRAM", program);
+        data.putExtra("CHANGE_PROGRAM_ENROLLMENT",enrollmentUid);
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/usescases/teiDashboard/teiProgramList/TeiProgramListActivity.java:105]: In here too.

File: app/src/androidTest/java/org/dhis2/usescases/teidashboard/TeiDashboardTest.kt [hunk 6]
```diff
@@ -293,9 +287,11 @@ class TeiDashboardTest : BaseTest() {
 
         enrollmentRobot {
             clickOnAProgramForEnrollment(womanProgram)
+            waitToDebounce(1000)
```
Comment (JaimeToca) [inline app/src/androidTest/java/org/dhis2/usescases/teidashboard/TeiDashboardTest.kt:290]: Why is this needed?

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureFragment/EventCaptureFormFragment.java (hunk 1 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureFragment/EventCaptureFormFragment.java:210]: it's a view type in the adapter

File: app/src/androidTest/java/org/dhis2/usescases/teidashboard/TeiDashboardTest.kt (hunk 6 above)
Comment (Balcan) [reply app/src/androidTest/java/org/dhis2/usescases/teidashboard/TeiDashboardTest.kt:290]: Not needed

**Comment by JaimeToca** [comment]
@Sharmyn28 Please, check out this PR since :). There were some changes regarding teidashboard intents.

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/edittext/EditTextCustomHolder.java [hunk 7]
```diff
@@ -60,7 +60,7 @@
                 if (isSearchMode || valueHasChanged()) {
                     sendAction();
                 }
-                closeKeyboard(binding.customEdittext.getEditText());
+//                closeKeyboard(binding.customEdittext.getEditText());
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/fields/edittext/EditTextCustomHolder.java:63]: ???
